test/Rx/Functional/FunctionalTestCase.php

```diff
@@ -99,4 +112,72 @@ protected function createTestScheduler()
     {
         return new TestScheduler();
     }
+
+    protected function convertMarblesToMessages(string $marbles, array $eventMap = [], \Exception $customError = null, $subscribePoint = 0)
```

Comment:
nit: perhaps extract this to some util? It might be useful in other contexts as well. E.g. demos and auto extracting marbles/tests from docs at some point.